### PR TITLE
Debt/MET-4655 Fix integration tests in dataset controller

### PR DESCRIPTION
**Pull request by JoanaCMS**

### CODE REVIEW

File: src/test/java/eu/europeana/metis/sandbox/test/utils/S3ContainerInitializerIT.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import org.testcontainers.utility.DockerImageName;
+
+import static org.testcontainers.containers.localstack.LocalStackContainer.Service.S3;
+
+public class S3ContainerInitializerIT {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(S3ContainerInitializerIT.class);
+    static final LocalStackContainer s3Container;
+    static final AmazonS3 s3Client;
+    public static final String S3_VERSION = "localstack/localstack:1.1.0";
```
Comment (jeortizquan) [inline src/test/java/eu/europeana/metis/sandbox/test/utils/S3ContainerInitializerIT.java:18]: We checked this version with Simon, for S3 compatible testcontainer and is good.

Comment (jeortizquan) [reply src/test/java/eu/europeana/metis/sandbox/test/utils/S3ContainerInitializerIT.java:18]: just add
```
private static final LocalStackContainer s3Container;
private static final AmazonS3 s3Client;
```

File: src/test/java/eu/europeana/metis/sandbox/test/utils/SolrContainerInitializerIT.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.testcontainers.containers.SolrContainer;
+import org.testcontainers.utility.DockerImageName;
+
+public class SolrContainerInitializerIT {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SolrContainerInitializerIT.class);
+    static final SolrContainer solrContainer;
```
Comment (jeortizquan) [inline src/test/java/eu/europeana/metis/sandbox/test/utils/SolrContainerInitializerIT.java:12]: private

File: src/test/java/eu/europeana/metis/sandbox/test/utils/SolrContainerInitializerIT.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    public static final String SOLR_VERSION = "solr:9.0.0";
```
Comment (jeortizquan) [inline src/test/java/eu/europeana/metis/sandbox/test/utils/SolrContainerInitializerIT.java:13]: we checked the version we are using at the moment, please modify it to `7.7.3`

File: pom.xml [hunk 4]
```diff
@@ -292,6 +292,24 @@
...
+      <artifactId>solr</artifactId>
+      <version>${version.testcontainers}</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.testcontainers</groupId>
+      <artifactId>localstack</artifactId>
+      <version>${version.testcontainers}</version>
+      <scope>test</scope>
+    </dependency>
```
Comment (jeortizquan) [inline pom.xml:312]: after reviewing and discussed, please update testcontainers version to `1.17.3`

File: src/test/java/eu/europeana/metis/sandbox/controller/DatasetControllerIT.java [hunk 5]
```diff
@@ -2,20 +2,26 @@
...
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
+import com.jayway.awaitility.Awaitility;
 import eu.europeana.metis.sandbox.SandboxApplication;
 import eu.europeana.metis.sandbox.common.locale.Country;
 import eu.europeana.metis.sandbox.common.locale.Language;
-import eu.europeana.metis.sandbox.test.utils.PostgresContainerInitializerIT;
-import eu.europeana.metis.sandbox.test.utils.RabbitMQContainerInitializerIT;
+import eu.europeana.metis.sandbox.test.utils.*;
+
```
Comment (jeortizquan) [inline src/test/java/eu/europeana/metis/sandbox/controller/DatasetControllerIT.java:15]: please remove the import `.*` and include the specific ones.

```
import eu.europeana.metis.sandbox.test.utils.MongoDBContainerInitializerIT;
import eu.europeana.metis.sandbox.test.utils.PostgresContainerInitializerIT;
import eu.europeana.metis.sandbox.test.utils.RabbitMQContainerInitializerIT;
import eu.europeana.metis.sandbox.test.utils.S3ContainerInitializerIT;
import eu.europeana.metis.sandbox.test.utils.SolrContainerInitializerIT;
```

File: src/test/java/eu/europeana/metis/sandbox/test/utils/MongoDBContainerInitializerIT.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.testcontainers.containers.MongoDBContainer;
+
+public class MongoDBContainerInitializerIT {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MongoDBContainerInitializerIT.class);
+    static final MongoDBContainer mongoDBContainer;
```
Comment (jeortizquan) [inline src/test/java/eu/europeana/metis/sandbox/test/utils/MongoDBContainerInitializerIT.java:11]: private

**Review by jeortizquan — CHANGES_REQUESTED** [review]
There are some version updates to do, and suggestions to the code to do. See the comments.

File: pom.xml (hunk 4 above)
Comment (JoanaCMS) [reply pom.xml:312]: Updated

File: src/test/java/eu/europeana/metis/sandbox/test/utils/SolrContainerInitializerIT.java (hunk 3 above)
Comment (JoanaCMS) [reply src/test/java/eu/europeana/metis/sandbox/test/utils/SolrContainerInitializerIT.java:13]: Updated

File: src/test/resources/application.yml [hunk 7]
```diff
@@ -70,32 +70,32 @@ sandbox:
       pool-size: 10
     predefinedSchemas:
       EDM-INTERNAL:
-        url:
+        url: http://ftp.eanadev.org/schema_zips/europeana_schemas-20220513.zip
         rootLocation: EDM-INTERNAL.xsd
         schematronLocation: schematron/schematron-internal.xsl
       EDM-EXTERNAL:
-        url:
+        url: http://ftp.eanadev.org/schema_zips/europeana_schemas-20220513.zip
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply src/test/resources/application.yml:77]: Thanks for pointing it out, I'll update it

File: src/test/resources/application.yml [hunk 8]
```diff
@@ -70,32 +70,32 @@ sandbox:
       pool-size: 10
     predefinedSchemas:
       EDM-INTERNAL:
-        url:
+        url: http://ftp.eanadev.org/schema_zips/europeana_schemas-20220513.zip
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply src/test/resources/application.yml:73]: I'll update it

File: src/test/java/eu/europeana/metis/sandbox/SandboxApplicationIT.java [hunk 9]
```diff
@@ -2,10 +2,11 @@
 
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 
-import eu.europeana.metis.sandbox.test.utils.PostgresContainerInitializerIT;
-import eu.europeana.metis.sandbox.test.utils.RabbitMQContainerInitializerIT;
+import eu.europeana.metis.sandbox.test.utils.*;
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply src/test/java/eu/europeana/metis/sandbox/SandboxApplicationIT.java:5]: This is done automatically so I don't always notice, but thanks for pointing it out! I'll update it

File: src/test/java/eu/europeana/metis/sandbox/controller/DatasetControllerIT.java (hunk 5 above)
Comment (JoanaCMS) [reply src/test/java/eu/europeana/metis/sandbox/controller/DatasetControllerIT.java:15]: Updated

File: src/test/java/eu/europeana/metis/sandbox/test/utils/S3ContainerInitializerIT.java (hunk 1 above)
Comment (JoanaCMS) [reply src/test/java/eu/europeana/metis/sandbox/test/utils/S3ContainerInitializerIT.java:18]: Done

File: src/test/java/eu/europeana/metis/sandbox/test/utils/SolrContainerInitializerIT.java (hunk 2 above)
Comment (JoanaCMS) [reply src/test/java/eu/europeana/metis/sandbox/test/utils/SolrContainerInitializerIT.java:12]: Done

File: src/test/java/eu/europeana/metis/sandbox/test/utils/MongoDBContainerInitializerIT.java (hunk 6 above)
Comment (JoanaCMS) [reply src/test/java/eu/europeana/metis/sandbox/test/utils/MongoDBContainerInitializerIT.java:11]: Done

File: src/test/java/eu/europeana/metis/sandbox/SandboxApplicationIT.java (hunk 9 above)
[the remark this answers is not in the document]
Comment (jeortizquan) [reply src/test/java/eu/europeana/metis/sandbox/SandboxApplicationIT.java]: Correct. Check please if the codestyle is loaded in the IDE; it should automatically expand this ones `.*`

**Review by jeortizquan — CHANGES_REQUESTED** [review]
I just noticed something today, that I didn't noticed last time and after that update we are ready.

File: src/test/java/eu/europeana/metis/sandbox/test/utils/S3ContainerInitializerIT.java [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+                .withCredentials(s3Container.getDefaultCredentialsProvider())
+                .build();
+        s3Client.createBucket(BUCKET_NAME);
+
+        logConfiguration();
+    }
+
+    private static void logConfiguration() {
+        LOGGER.info("S3 service container created:");
+        LOGGER.info("Access Key Id: {}", s3Container.getAccessKey());
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply src/test/java/eu/europeana/metis/sandbox/test/utils/S3ContainerInitializerIT.java:38]: Thanks for pointing it out, you're right, it is best to remove the access key ans secret key. 
I've put them at first for testing purposes but I'll remove them
The other 2 two I think they're fine

**Review by jeortizquan — APPROVED** [review]

**Pull request merged**